src/devices/vdml_motors.cpp

```diff
@@ -17,265 +17,662 @@ namespace pros {
-	return motor_move(_port, voltage);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_move(_ports[i], voltage);
+	}
+	return motor_move(_ports[0], voltage);
 }
 
-std::int32_t Motor::move_absolute(const double position, const std::int32_t velocity) const {
+std::int32_t Motor_Group::move_absolute(const double position, const std::int32_t velocity) const {
+	empty_motor_group_check(PROS_ERR);
 	push_motor_configuration();
-	return motor_move_absolute(_port, position, velocity);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_move_absolute(_ports[i], position, velocity);
+	}
+	return motor_move_absolute(_ports[0], position, velocity);
 }
 
-std::int32_t Motor::move_relative(const double position, const std::int32_t velocity) const {
+std::int32_t Motor_Group::move_relative(const double position, const std::int32_t velocity) const {
+	empty_motor_group_check(PROS_ERR);
 	push_motor_configuration();
-	return motor_move_relative(_port, position, velocity);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_move_relative(_ports[i], position, velocity);
+	}
+	return motor_move_relative(_ports[0], position, velocity);
 }
 
-std::int32_t Motor::move_velocity(const std::int32_t velocity) const {
+std::int32_t Motor_Group::move_velocity(const std::int32_t velocity) const {
+	empty_motor_group_check(PROS_ERR);
 	push_motor_configuration();
-	return motor_move_velocity(_port, velocity);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_move_velocity(_ports[i], velocity);
+	}
+	return motor_move_velocity(_ports[0], velocity);
 }
 
-std::int32_t Motor::move_voltage(const std::int32_t voltage) const {
+std::int32_t Motor_Group::move_voltage(const std::int32_t voltage) const {
+	empty_motor_group_check(PROS_ERR);
 	push_motor_configuration();
-	return motor_move_voltage(_port, voltage);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_move_voltage(_ports[i], voltage);
+	}
+	return motor_move_voltage(_ports[0], voltage);
 }
 
-std::int32_t Motor::brake(void) const {
+std::int32_t Motor_Group::brake(void) const {
+	empty_motor_group_check(PROS_ERR);
 	push_motor_configuration();
-	return motor_brake(_port);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_brake(_ports[i]);
+	}
+	return motor_brake(_ports[0]);
 }
 
-std::int32_t Motor::modify_profiled_velocity(const std::int32_t velocity) const {
+std::int32_t Motor_Group::modify_profiled_velocity(const std::int32_t velocity) const {
+	empty_motor_group_check(PROS_ERR);
 	push_motor_configuration();
-	return motor_modify_profiled_velocity(_port, velocity);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_modify_profiled_velocity(_ports[i], velocity);
+	}
+	return motor_modify_profiled_velocity(_ports[0], velocity);
 }
 
-double Motor::get_actual_velocity(void) const {
+double Motor_Group::get_actual_velocity(void) const {
+	empty_motor_group_check(PROS_ERR_F);
 	push_motor_configuration();
-	return motor_get_actual_velocity(_port);
+	return motor_get_actual_velocity(_ports[0]);
 }
```

Comment:
The reason we are going for the design we did is that we thought it would be more confusing to users if we had a base class for motors. Instead we opted for the idea that motors are a "specialization" of the motor groups. This removes one extra class in the API, and means that we don't have to explain the abstract motor to those new to C++. 

I see your point about the single motor getters/setters. Maybe we could make them take an optional index with 0 as the default? Then again, that is kinda redundant with the operator[] method.

Reply:
Additional thought about my 2nd paragraph: users can use the "every" methods and index them for specific motors.

Reply:
> Instead we opted for the idea that motors are a "specialization" of the motor groups
see top level code review comment for my thoughts on this

> I see your point about the single motor getters/setters. Maybe we could make them take an optional index with 0 as the default?
maybe, but then its weird for the `Motor` objects that it takes a param. This whole hierarchy doesn't really work with how OOP hierarchy is meant to be.

> users can use the "every" methods and index them for specific motors
This is very bad for performance, every call creates a vector which needs to heap allocate which is expensive.  In reality, probably wont matter for 99% of users, but..